### ISSUE TITLE
Use the on-premises (local) version of Structurizr Express

### ISSUE THREAD

**Issue by aviflax** [FundingCircle/fc4-framework#288]
Because the Web-accessible version has been discontinued, as per in #268.

For more on the specific reasons to go with this approach, please see [this comment on #268](https://github.com/FundingCircle/fc4-framework/issues/268#issuecomment-592165632) and [this comment on #284](https://github.com/FundingCircle/fc4-framework/issues/284#issuecomment-641446349).

**Comment by arichiardi** [comment]
Following up on previous conversations I think that adding support for a local Structurizr Express makes a lot of sense not just to address the momentary issue with the online version but in general...when I first looked at this I was actually looking at something that I could use completely offline :smile:

**Comment by aviflax** [comment]
Thanks @arichiardi for the feedback!

If this CLI option would be useful to you right now, for the time being, I’d be happy to add it. Please confirm — I just want to make sure I don’t spend time on something that someone _might_ use.

As for the longer term, though, as per #284 my plan is to remove the support for / integration with Structurizr Express by the end of June; I plan to replace it with the new DSL / data structure for models and views and a new, fully integrated, encapsulated, local, and offline renderer. If you have any thoughts on that I’d love to hear them, but over in #284 if you don’t mind.

**Comment by arichiardi** [comment]
Hey @aviflax it makes sense and if I am the only one...well I would suggest to not spend any time unless very trivial. I am still in the convincing phase here in my company. You know how it works...

> As for the longer term, though, as per #284 my plan is to remove the support for / integration with Structurizr Express by the end of June; I plan to replace it with the new DSL / data structure for models and views and a new, fully integrated, encapsulated, local, and offline renderer. 

This sounds like a better plan and surely the timeline is not that bad. Thanks for sharing it is helpful.

**Comment by aviflax** [comment]
Got it. Thank you! 🤗

**Comment by aviflax** [comment]
I think I’ve run out of time to implement the plan in #284 as-is before the deadline, which is in 3 weeks. So for now, as a stopgap, I’ll try this.

**Comment by aviflax** [comment]
@simonbrowndotje I’m almost done with this work, at least locally. Now I’m trying to adapt FC4’s CI jobs to also use the on-prem version of Structurizr Express, and I’ve hit a blocker: I can’t figure out a way to programmatically download the `.war` file.

Here’s what I tried:

```shell
curl -v -O --anyauth --user "$USER:$PASS" https://structurizr.com/user/download/structurizr-onpremises-1979.war
```

I also tried `--basic` instead of `--anyauth`.

Sorry to bother you, but if you have any suggestions for accomplishing this, I’d be very grateful!

(I’d make sure that the CI job cached the download, so it shouldn’t add much load or bandwidth expense to your website.)

As an alternative, if you were to grant me permission to upload a copy of the `.war` file into a private, secure S3 bucket, I could have my CI job download the `.war` file securely from that bucket. I think that’d work quite well, but I wouldn’t feel comfortable doing so without your express permission.

Thank you!
